tests/__mocks__/psElevateTokenHandler.js

```diff
@@ -0,0 +1,257 @@
+
+/***
+ * @description Visualforce page used to handle the payment services credit card tokenization request
+ */
+const TOKENIZE_CARD_PAGE_NAME = 'GE_TokenizeCard';
+
+const MOUNT_IFRAME_EVENT_ACTION = 'mount';
+
+const SET_PAYMENT_METHOD_EVENT_ACTION = 'setPaymentMethod';
+
+/***
+ * @description Max number of ms to wait for the response containing a token or an error
+ */
+const TOKENIZE_TIMEOUT_MS = 10000;
+
+const NON_NAMESPACED_CHARACTER = 'c';
+
+
+/***
+ * @description Payment services Elevate credit card tokenization service
+ */
+class psElevateTokenHandler {
+    /***
+     * @description Custom labels
+     */
+    labels = Object.freeze({
+        tokenRequestTimedOut
+    });
+
+    _visualforceOriginUrls;
+    _visualforceOrigin;
+    _componentRef;
+
+    /***
+     * @description Returns credit card tokenization Visualforce page URL
+     */
+    getTokenizeCardPageURL() {
+        return this.currentNamespace
+            ? `/apex/${this.currentNamespace}__${TOKENIZE_CARD_PAGE_NAME}`
+            : `/apex/${TOKENIZE_CARD_PAGE_NAME}`;
+    }
+
+    getVisualForceOriginURLs(domainInfo, namespace) {
+        const url = `https://${domainInfo.orgDomain}--${namespace}.visualforce.com`;
+        const alternateUrl = `https://${domainInfo.orgDomain}--${namespace}.${domainInfo.podName}.visual.force.com`;
+        const productionEnhancedUrl = `https://${domainInfo.orgDomain}--${namespace}.vf.force.com`;
+        const sandboxEnhancedUrl =  `https://${domainInfo.orgDomain}--${namespace}.sandbox.vf.force.com`;
+
+        return [
+            {value: url},
+            {value: alternateUrl},
+            {value: productionEnhancedUrl},
+            {value: sandboxEnhancedUrl}
+        ];
+    }
+
+    /***
+     * @description Builds the Visualforce origin url that we need in order to
+     * make sure we're only listening for messages from the correct source.
+     */
+    setVisualforceOriginURLs(domainInfo) {
+        if (isNull(domainInfo)) {
+            return;
+        }
+        const namespace = this.currentNamespace
+            ? this.currentNamespace
+            : NON_NAMESPACED_CHARACTER;
+        this._visualforceOriginUrls =
+            this.getVisualForceOriginURLs(domainInfo, namespace);
+
+    }
+
+    /***
+     * @description Returns the NPSP namespace (if any)
+     */
+    get currentNamespace() {
+        return getNamespace(PAYMENT_AUTHORIZATION_TOKEN_FIELD.fieldApiName);
+    }
+
+    /***
+     * @description Dispatches the application event when the Elevate credit card iframe
+     * is displayed or hidden
+     */
+    dispatchApplicationEvent(eventName, payload) {
+        fireEvent(null, eventName, payload);
+    }
+
+    /***
+     * @description Listens for a message from the Visualforce iframe.
+     * Rejects any messages from an unknown origin.
+     */
+    registerPostMessageListener(component) {
+        const self = this;
+        this._componentRef = component;
```

Comment:
This is different from the original implementation. Since we're not using `window.onmessage` we need a reference to the original component in order to call `handleMessage` on it.